### PR TITLE
Fix global role count with standard and admin users

### LINKED ISSUE #7046: Remove a ROLE : The Confirmation window shows incorrect user count
**Setup**
- Rancher version: v2.6-2e0026699d7f3498979f9007a17b20668cb076ef-head
- Browser type & version: Chrome (Version 105.0.5195.125 ) on MAC

**Describe the bug**

While removing a ROLE, the user is getting a confirmation window. On the window the Caution in RED color - the 'number' of bounded users to the roles is incorrect. It is displaying the SUM of all the users counts from all the ROLES which are associate to the each role. It would be good to have the actual number of users bounded as a total.
Ex: Role 1 has 2 users (U1, U2)
Role 2 has users (U1, U3)

The dialog window is showing as 4 users instead of 3 users as U1 is common in both roles

Note: In the attached screen the total users shows 76 which leads some confusion as this node has just 8 users all together.

**To Reproduce**
**Steps:**

1. Create a User
2. Create a Role
3. Create a Cluster
4. Create a Project Name Space
5. Create a Project
6. Associate Users to multiple roles
7. Roles : Select multiple Roles (Global / Custom)
8. Try to delete multiple Roles and check the text 'Caution: There are 76 users bound to the role(s) about to be deleted. Do you want to proceed?'

**Actual Behavior:**
1. On the window the Caution in RED color - the 'number' of bounded users to the roles is incorrect. It is displaying the SUM of all the user counts from all the ROLES which are associate to the each role. It would be good to have the actual number of users bounded as a total.
2. In the attached screen the total users shows 76 which leads some confusion as this node has just 8 users all together.
	

**Expected Behavior:**
1. Appropriate bounded users should display instead of summing all the users counts
2. In the attached screen the total users shows 76 which leads some confusion as this node has just 8 users all together and it should say 8 users instead of 76 users.

image
rancher/dashboard#6578

### PR DESCRIPTION
**Pull request by jordojordo**
### Summary
Fixes #7046 

### Occurred changes and/or fixed issues

This fixes the issue of counting Global Roles with the Standard User option.
The Standard User global role binding includes not only the standard users, but also duplicates of the admin accounts. So to fix this we now fetch the users to filter against the count of role bindings found.

### Areas or cases that should be tested

- Create a few standard users and admin users
- Attempt to delete the Administrator and Standard User global roles
- Notice the count will contain the exact number of actual users  

### Screenshot/Video

__6 Users__
6-users

__Before - 8 users total__
before-8-users-bound

__After - 6 users total__
after-6-users-bound

### CODE REVIEW

File: shell/promptRemove/mixin/roleDeletionCheck.js [hunk 1]
```diff
@@ -66,12 +66,26 @@ export default {
...
         }, { root: true });
 
+        // We need to fetch the users here in order to get an accurate count when selecting global roles.
+        const users = await this.$store.dispatch('management/request', {
+          url:           `/v1/${ MANAGEMENT.USER }`,
+          method:        'get',
+        }, { root: true });
+
+        const userMap = users.data?.reduce((map, user) => {
+          if ( user.username ) {
```
Comment (richard-cox) [inline shell/promptRemove/mixin/roleDeletionCheck.js:76]: I'm not 100% sure, but i think what you're after here is to exclude system accounts which could be done via `user.isSystem`? For this user case maybe the existing `user.username` is better?

**Review by richard-cox — COMMENTED** [review]
There's a bit of a omission generally with this feature - Groups. Roles are bound to either users or groups (see `groupPrincipalName` instead of `userName` for the role binding). Is that tracked somewhere already, if not i'll create an issue?

File: shell/promptRemove/mixin/roleDeletionCheck.js (hunk 1 above)
Comment (jordojordo) [reply shell/promptRemove/mixin/roleDeletionCheck.js:76]: Yes exactly, system accounts was the term I was looking for. I'm not entirely sure if using `user.isSystem` would be much better than searching for the username as `user.username` only exists if it isn't a system account.

**Comment by jordojordo** [comment]
I don't believe this is tracked anywhere. When I look at the data returned from a role binding resource I'm not seeing any mention of `groupPrincipalName`, but maybe I'm looking in the wrong place?

File: shell/promptRemove/mixin/roleDeletionCheck.js (hunk 1 above)
Comment (richard-cox) [reply shell/promptRemove/mixin/roleDeletionCheck.js:76]: Let's keep as is for the moment. `isSystem` was brought in by Vince to do I think the exact task as here, but given  everything is tested and working lets get this in and re-visit if there's issues

Comment (nwmac) [reply shell/promptRemove/mixin/roleDeletionCheck.js:76]: @richard-cox Agree - I think this is okay

**Comment by nwmac** [comment]
@richard-cox Can this go in? Sounds like there is follow-on work - but this at least addresses the issue from QA?

**Review by richard-cox — APPROVED** [review]
I've created https://github.com/rancher/dashboard/issues/7199 to cover group warnings.

The properties on the binding exist dependent on the type bound to, so the group one will only show if bound to a group

**Pull request merged**